lib/widgets/buttons.dart

```diff
@@ -417,3 +418,57 @@ class CopyButton extends StatelessWidget {
     );
   }
 }
+
+class FavoriteButton extends StatelessWidget {
+  const FavoriteButton({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = ThemeControl.instance.theme;
+    return StreamBuilder(
+      stream: ContentControl.instance.onContentChange,
+      builder: (context, snapshot) {
+        final currentSong = PlaybackControl.instance.currentSong;
+        return HeartButton(
+          active: currentSong.isFavorite,
```

Comment:
```suggestion
    return StreamBuilder<Song>(
      stream: PlaybackControl.instance.onSongChange,
      builder: (context, snapshot) {
        final currentSong = snapshot.data;
        return HeartButton(
          active: currentSong?.isFavorite ?? false,
```
This caused favorites in the to not show the correct value sometimes.

Reply:
Yes, thanks. We still need to have the `onContentChange` here though, since otherwise the value of `isFavorite` won't update